IdentityCore/src/util/NSData+JWT.m

```diff
@@ -0,0 +1,126 @@
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "NSData+JWT.h"
+#import <CommonCrypto/CommonDigest.h>
+#import <Security/Security.h>
+#import <Security/SecKey.h>
+
+@implementation NSData (JWT)
+
+#if TARGET_OS_IPHONE
+
+- (NSData *)signHashWithPrivateKey:(SecKeyRef)privateKey
+{
+    NSData *signedHash = nil;
+    size_t signedHashBytesSize = SecKeyGetBlockSize(privateKey);
+    uint8_t *signedHashBytes = calloc(signedHashBytesSize, 1);
+
+    if (!signedHashBytes)
+    {
+        return nil;
+    }
+
+    OSStatus status = errSecAuthFailed;
+
+    status = SecKeyRawSign(privateKey,
+                           kSecPaddingPKCS1SHA256,
+                           [self bytes],
+                           CC_SHA256_DIGEST_LENGTH,
+                           signedHashBytes,
+                           &signedHashBytesSize);
+
+    if (status != errSecSuccess)
+    {
+        MSID_LOG_ERROR(nil, @"Failed to sign JWT %d", (int)status);
+        free(signedHashBytes);
+        return nil;
+    }
+
+    signedHash = [NSData dataWithBytes:signedHashBytes
+                                length:(NSUInteger)signedHashBytesSize];
+
+    free(signedHashBytes);
+    return signedHash;
+}
+
+#else
+
+- (NSData *)signHashWithPrivateKey:(SecKeyRef)privateKey
+{
+    CFErrorRef error = nil;
+    // Create signer
+    SecTransformRef signer = SecSignTransformCreate(privateKey, &error);
+
+    if (!signer)
+    {
+        MSID_LOG_ERROR(nil, @"Failed to sign JWT %@", error);
+        return nil;
+    }
+
+    BOOL result = YES;
+    // Set attributes
+    result &= [self setAttributeOnSigner:signer attributeKey:kSecPaddingKey attributeValue:kSecPaddingPKCS1Key];
+    result &= [self setAttributeOnSigner:signer attributeKey:kSecInputIsAttributeName attributeValue:kSecInputIsDigest];
+    result &= [self setAttributeOnSigner:signer attributeKey:kSecTransformInputAttributeName attributeValue:(__bridge CFDataRef)self];
+    result &= [self setAttributeOnSigner:signer attributeKey:kSecDigestTypeAttribute attributeValue:kSecDigestSHA2];
+    result &= [self setAttributeOnSigner:signer attributeKey:kSecDigestLengthAttribute attributeValue:(__bridge CFNumberRef)@(256)];
+
+    if (!result)
+    {
+        MSID_LOG_ERROR(nil, @"Failed to sign JWT %@", error);
+        CFRelease(signer);
+        return nil;
+    }
+
+    CFDataRef resultData = SecTransformExecute(signer, &error);
+
```

Comment:
nit: you can release signer here and remove releasing from lines 101 & 105.

Reply:
Fixed :)